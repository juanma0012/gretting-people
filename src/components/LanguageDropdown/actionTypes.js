export const SET_LANGUAGE = 'languageDropdown/SET_LANGUAGE';
