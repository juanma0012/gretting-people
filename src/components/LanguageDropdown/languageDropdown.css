.languageDropdown > select {
    width: 100%;
    font-size: 1em;
	margin: 2px;
	color: #535353;
	padding: 15px 5px 15px 25px;
	height: 50px;
	cursor: pointer;
}

.languageDropdown > select:hover {
	background-color: #dbe2e8;
	color: #7b7b7b;
}
