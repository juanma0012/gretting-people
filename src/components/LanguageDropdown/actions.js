import * as actionTypes from './actionTypes';
import * as greetingActions from '../GreetingList/actions';

export const setLanguage = (language) => {
    return (dispatch, getState) => {
        dispatch({ type: actionTypes.SET_LANGUAGE, language });
        let state = getState();
        let people = state.peopleReducer.people;
        dispatch(greetingActions.setGreetings(people, language));
    }
};
