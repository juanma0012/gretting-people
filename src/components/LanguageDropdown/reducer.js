import { resolveEach } from 'redux-resolver';
import { updateObject, defaultLanguage } from '../../shared/utility';
import * as actionTypes from './actionTypes';

const initialState = {
    language: defaultLanguage
};

const setLanguage = (state, action) => {
    return updateObject(state, {
        language: action.language
    });
};

export default resolveEach(initialState, {
    [actionTypes.SET_LANGUAGE]: setLanguage
});
