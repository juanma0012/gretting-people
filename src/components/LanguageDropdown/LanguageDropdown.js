import { connect } from 'react-redux'
import React from 'react'
import PropTypes from 'prop-types'
import { greetingByLanguage, toCapitalLetter } from '../../shared/utility';
import * as actions from './actions'
import './languageDropdown.css';

export const LanguageDropdown = ({ onSetLanguage }) => {
    let select;
    let languages = Object.keys(greetingByLanguage);
    const languagesList = languages.map((language, i) => (
        <option value={language} key={i}>{toCapitalLetter(language)}</option>
    ));
    const onChangeLanguage = e => onSetLanguage(e.target.value);

    return (
        <div className="languageDropdown">
            <select value={select} onChange={onChangeLanguage}>
                {languagesList}
            </select>
        </div>
    )
};

LanguageDropdown.propTypes = {
    onSetLanguage: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
    onSetLanguage: (people) => dispatch(actions.setLanguage(people))
});

export default connect(
    null,
    mapDispatchToProps
)(LanguageDropdown);
