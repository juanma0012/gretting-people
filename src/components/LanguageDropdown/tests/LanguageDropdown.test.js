import React from 'react';
import { shallow } from 'enzyme';
import { LanguageDropdown } from '../LanguageDropdown';

describe('Component LanguageDropdown ', () => {
    describe('WHEN is initialized ', () => {
        it('SHOULD render correctly ', () => {
            const component = shallow(<LanguageDropdown onSetLanguage={() => { }} />);
            expect(component).toMatchSnapshot();
        });
    });

    describe('WHEN a language is changed in the dropdown ', () => {
        const changeFn = jest.fn();
        it('SHOULD calls the method  onSetLanguage', () => {
            const component = shallow(<LanguageDropdown onSetLanguage={changeFn} />);
            component.find('select').simulate('change', { target: { value: "spanish" } });
            expect(changeFn).toHaveBeenCalled();
        });
    });
});
