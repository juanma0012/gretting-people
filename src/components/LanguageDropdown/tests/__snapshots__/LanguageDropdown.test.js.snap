// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LanguageDropdown  WHEN is initialized  SHOULD render correctly  1`] = `
<div
  className="languageDropdown"
>
  <select
    onChange={[Function]}
  >
    <option
      key="0"
      value="english"
    >
      English
    </option>
    <option
      key="1"
      value="spanish"
    >
      Spanish
    </option>
    <option
      key="2"
      value="french"
    >
      French
    </option>
    <option
      key="3"
      value="portuguese"
    >
      Portuguese
    </option>
    <option
      key="4"
      value="italian"
    >
      Italian
    </option>
  </select>
</div>
`;
